core/dbt/parser/partial.py

```diff
@@ -630,6 +646,31 @@ def delete_fixture_node(self, source_file):
             source_file.unit_tests.remove(unique_id)
         self.saved_manifest.files.pop(source_file.file_id)
 
+    def delete_function_node(self, source_file: SourceFile):
+        # There should always be a node for a Function file
+        if not isinstance(source_file, SourceFile) or not source_file.functions:
+            return
+
+        # There can only be one node of a function
+        function_unique_id = source_file.functions[0]
+
+        # Remove the function node from the saved manifest
+        function_node = self.saved_manifest.functions.pop(function_unique_id)
+
+        # Remove the function node from the source file so that it's not viewed as a
+        # duplicate when it's re-added
+        source_file.nodes.remove(function_unique_id)
+
+        # schedule function for parsing.
+        # Note: We _don't_ need to schedule referencing nodes for reparsing, because a change in a function changes nothing
```

Comment:
If the number of parameters has changed, is that something that's detected at parse time? Or only compilation time? If it's detected at parse time, then we should schedule referencing nodes in order to get that error. If it's not detected at parse time, this is fine.

Reply:
I've added a test that confirms reparsing of the downstreams is not required if the 3 part name of the function changes